### ISSUE TITLE
Add support for multiple artifact extensions

### ISSUE THREAD

**Issue by scottTomaszewski** [jenkinsci/artifact-promotion-plugin#11]
In configuration, it would be nice to be able to promote multiple artifacts with the same groupId, artifactId, version, etc.  For example, to promote a pom and jar:

```
job {
    steps {
        artifactPromotion {
          groupId("foo")
          artifactId("bar")
          version("1.2.3")
          extension(["pom", "jar"])
          stagingRepository("nexus", "foo", "bar")
          releaseRepository("nexus", "foo", "bar")
          debug(false)
        }
    }
}
```

**Comment by hcguersoy** [comment]
Thank you, I'll take a look on it how this can be implemented. 
Feel free to implement it by your own and open a pull request :-)

**Comment by D0rmouse** [comment]
@hcguersoy Why not simply promote all artifacts for the specified version? And optionally and exclusion matcher or something?

Now I need to call `artifactPromotion` separately for every type of extension :( 
And then the same for every classifier :(

**Comment by hcguersoy** [comment]
> @hcguersoy Why not simply promote all artifacts for the specified version? And optionally an exclusion matcher or something?
> 
> Now I need to call `artifactPromotion` separately for every type of extension :(
> And then the same for every classifier :(

@D0rmouse yes, would be a possibility and usable for most of the projects.
Can you provide a PR?

**Comment by D0rmouse** [comment]
@hcguersoy Not sure my company will give me more time to spend on this. I may have to drop it or find alternate solutions.
